lib/src/linter.dart

```diff
@@ -154,7 +154,9 @@ class LinterOptions extends DriverOptions {
     }
   }
   void configure(LintConfig config) {
-    enabledLints = ruleRegistry.enabled(config);
+    // TODO(pquitslund): revisit these default-to-on semantics.
+    enabledLints = ruleRegistry.where((LintRule rule) =>
+        !config.ruleConfigs.any((rc) => rc.disables(rule.name)));
```

Comment:
Nope.  That was deliberate.  The current semantics are actually I think **wrong**.  I'll clean that up in a follow-up CL.